src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.LeadingZeroCount.cs

```diff
@@ -22,13 +28,56 @@ public static void LeadingZeroCount<T>(ReadOnlySpan<T> x, Span<T> destination)
             InvokeSpanIntoSpan<T, LeadingZeroCountOperator<T>>(x, destination);
 
         /// <summary>T.LeadingZeroCount(x)</summary>
-        internal readonly struct LeadingZeroCountOperator<T> : IUnaryOperator<T, T> where T : IBinaryInteger<T>
+        internal readonly unsafe struct LeadingZeroCountOperator<T> : IUnaryOperator<T, T> where T : IBinaryInteger<T>
         {
-            public static bool Vectorizable => false; // TODO: Vectorize
+            public static bool Vectorizable =>
+                (Avx512CD.VL.IsSupported && (sizeof(T) == 4 || sizeof(T) == 8)) ||
+                (AdvSimd.IsSupported && (sizeof(T) == 1 || sizeof(T) == 2 || sizeof(T) == 4));
+
             public static T Invoke(T x) => T.LeadingZeroCount(x);
-            public static Vector128<T> Invoke(Vector128<T> x) => throw new NotSupportedException();
-            public static Vector256<T> Invoke(Vector256<T> x) => throw new NotSupportedException();
-            public static Vector512<T> Invoke(Vector512<T> x) => throw new NotSupportedException();
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Vector128<T> Invoke(Vector128<T> x)
+            {
+                if (Avx512CD.VL.IsSupported)
+                {
+                    if (sizeof(T) == 4) return Avx512CD.VL.LeadingZeroCount(x.AsUInt32()).As<uint, T>();
+                    if (sizeof(T) == 8) return Avx512CD.VL.LeadingZeroCount(x.AsUInt64()).As<ulong, T>();
+                }
+
+                Debug.Assert(AdvSimd.IsSupported);
```

Comment:
We want these to be explicit. It makes it work better with the trimmer for other platforms and matches the requirements we have for corelib around using intrinsics (this isn't in corelib, but its still good to follow those practices as its required for special crossgen handling to ever be usable for the assembly).

Reply:
The rest of TP is like this, as I'd previously addressed feedback around throws. I suggest this be merged as is, and if we want to change the approach, it be done across all operators en mass together.